internal/ui/backup/text.go

```diff
@@ -186,3 +186,25 @@ func (b *TextProgress) Finish(snapshotID restic.ID, start time.Time, summary *Su
 		formatDuration(time.Since(start)),
 	)
 }
+
+// Return finishing stats in a struct.
+func (b *TextProgress) FinishSummary(snapshotID restic.ID, start time.Time, summary *Summary, dryRun bool) summaryOutput {
```

Comment:
This function does not require any data stored in the TextProgress struct. Thus please extract it into a separate function and remove the duplication with the JsonProgress.